Merge main from dependabot/dependabot-core

GiriB opened the pull request:
rebase the forked repository with dependaot/dependabot-core:main branch

feelepxyz committed:
npm: handle latest version requirement

Handle `latest` version requirements which are valid in npm:

```
"dependencies": {
  "etag" : "latest"
}
```

feelepxyz committed:
Update CHANGELOG.md

feelepxyz committed:
Merge pull request #3787 from dependabot/v0.148.7-release-notes

v0.148.7

jurre committed:
v0.148.8

jurre committed:
Merge pull request #3788 from dependabot/v0.148.8-release-notes

v0.148.8

feelepxyz committed:
Check version is correct before filtering ignored

This is yet another patch to handle edge-cases around filtering ignored
versions where the dependency is a git dependency and the version has
been initialised as a sha and therefore can't be used to filter ignored
versions.

xlgmokha committed:
Merge remote-tracking branch 'up/main' into updates-1480-source-address-hostname

xlgmokha committed:
fix: remove duplicate method definition

xlgmokha committed:
test: check if version constraint is fixed

xlgmokha committed:
fix: use nil version for pessimistic constraints

xlgmokha committed:
test: detect resolved version from the lockfile

xlgmokha committed:
fix: parse version from terraform lock file

xlgmokha committed:
refactor: collapse methods

xlgmokha committed:
style: fix linter errors

xlgmokha committed:
style: collapse lines

xlgmokha committed:
test: add spec to parse provider metadata from a custom registry

xlgmokha committed:
fix: remove check for public hostname

xlgmokha committed:
style: remove unused let

xlgmokha committed:
test: fetch providers versions from a custom registry

xlgmokha committed:
fix: remove restriction to use registry.terraform.io

xlgmokha committed:
test: fetch module versions from a custom registry

xlgmokha committed:
style: fix linter errors

xlgmokha committed:
fix: remove restriction to use registry.terraform.io

xlgmokha committed:
refactor: make registry.terraform.io the default

xlgmokha committed:
refactor: forward credentials to the registry client

xlgmokha committed:
fix: inject Authorization header

Nishnha committed:
Add lockfile to the fetched_files array in FileFetcher

xlgmokha committed:
Merge pull request #3756 from xlgmokha/updates-1480-source-address-hostname

Parse version from terraform lock file

xlgmokha committed:
Merge remote-tracking branch 'up/main' into updates-1480-metadata

Nishnha committed:
Add lockfile to updated_dependency_files array in FileUpdater

Nishnha committed:
Add tests for versions.tf dependency updates with and without a lockfile

xlgmokha committed:
test: remove test that has been replaced

Nishnha committed:
Update hashicorp/aws to a valid version in file_parser_spec test

dependabot[bot] committed:
build(deps-dev): bump jest in /npm_and_yarn/helpers

Bumps [jest](https://github.com/facebook/jest) from 26.6.3 to 27.0.1.
- [Release notes](https://github.com/facebook/jest/releases)
- [Changelog](https://github.com/facebook/jest/blob/master/CHANGELOG.md)
- [Commits](https://github.com/facebook/jest/compare/v26.6.3...v27.0.1)

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Terraform: Do not set dependency.version for version ranges

When a dependency specifies a version range, we do not want to set the
dependency objects version, as we cannot reliably determine it. Instead,
we need to rely on the _requirement_. For example, `~> 2.0` is not a
valid _version_, so when parsing it, we should set the `version`
attribute to nil.

This was already handled for terraform modules, but not for providers.

feelepxyz committed:
Merge pull request #3789 from dependabot/feelepxyz/check-version-correct-ignored-versions

Check version is correct before filtering ignored

feelepxyz committed:
Merge pull request #3791 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/jest-27.0.1

build(deps-dev): bump jest from 26.6.3 to 27.0.1 in /npm_and_yarn/helpers

feelepxyz committed:
Merge pull request #3781 from dependabot/feelepxyz/fix-npm-latest-requirement

npm: handle latest version requirement

jurre committed:
Merge pull request #3792 from dependabot/jurre/handle-provider-version-ranges

Terraform: Do not set dependency.version for version ranges

jurre committed:
v0.148.9

feelepxyz committed:
Composer: fix git clone error in lockfile updater

Fix error matching for `GitDependenciesNotReachable` in the lockfile
updater.

We're still seeing the error I tried to fix in the version resolver:
https://github.com/dependabot/dependabot-core/pull/3779

It looks like we need to match the same error in the lockfile updater as
it might raise for a dependency we're not trying to update.

Sharing the constant doesn't feel ideal. It would be better to extract
the error handling to something we can share between the lockfile
updater and version resolver but wanted to get this fix in to improve
our error rate.

jurre committed:
Merge pull request #3794 from dependabot/v0.148.9-release-notes

v0.148.9

Nishnha committed:
Run shell commands with escaped arguments

Co-authored-by: Philip Harrison <philip@mailharrison.com>

feelepxyz committed:
Yarn: use .yarnrc file if present

Fixes a bug when trying to force yarn to use `registry.npmjs.org`
instead of `registry.yarnpkg.com`.

Previously if a .yarnrc file was present with registry `registry
"https://registry.npmjs.org"` it would be ignored during updates as we
run yarn programtically and didn't instruct it to use the config file.

I've also tested a dry-run without a `.yarnrc` file present.

feelepxyz committed:
Merge pull request #3796 from dependabot/feelepxyz/yarn-use-yarnrc-files-if-present

Yarn: use .yarnrc file if present

feelepxyz committed:
v0.148.10

jurre committed:
Bundler: Update bundler to 2.2.18

https://github.com/rubygems/rubygems/blob/master/bundler/CHANGELOG.md#2218-may-25-2021

feelepxyz committed:
Merge pull request #3797 from dependabot/v0.148.10-release-notes

v0.148.10

xlgmokha committed:
Merge pull request #3790 from xlgmokha/updates-1480-metadata

feat(Terraform): Inject authorization header

xlgmokha committed:
v0.149.0

xlgmokha committed:
Merge pull request #3799 from xlgmokha/v0.149.0-release-notes

v0.149.0

jurre committed:
Bundler: Fix ruby version patch for 2.2.18

Bundler 2.2.18 no longer loads all specs from remote sources when
building the source requirements. This is where we used to hook into to
patch up any required ruby version before building the source
requirements.

This changes the patch to prepend this behavior to the
`source_requirements` method.

mctofu committed:
gomod: UpdateChecker - handle invalid module path error on update

jurre committed:
Merge pull request #3798 from dependabot/jurre/bundler-2.2.18

jurre committed:
v0.149.1

jurre committed:
Tests: avoid squatted repositories

A few repositories we expect not to exist on npm have been squatted.
Where possible this updates those tests to use a namespace that we own,
and for one fixture we explicitly tested against an unscoped package,
those tests have been removed as we do not expect behavior to differ
between scoped and unscoped packages.

jurre committed:
Merge pull request #3801 from dependabot/v0.149.1-release-notes

v0.149.1

jurre committed:
Merge pull request #3802 from dependabot/jurre/fix-npm-test-failures

Tests: avoid squatted repositories

dependabot[bot] committed:
Bump eslint from 7.26.0 to 7.27.0 in /npm_and_yarn/helpers

Bumps [eslint](https://github.com/eslint/eslint) from 7.26.0 to 7.27.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.26.0...v7.27.0)

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
v0.149.2

jurre committed:
Merge pull request #3803 from dependabot/v0.149.2-release-notes

v0.149.2

mctofu committed:
Merge pull request #3800 from dependabot/mctofu/module-path-change-on-update

gomod: UpdateChecker - handle invalid module path error on update

Nishnha committed:
Inline expected values for lockfile tests

Nishnha committed:
Fix FileUpdater test version and name issues

Nishnha committed:
Update FileFetcher#lock_file and FileSelector#lock_file to use FileSelector#lock_file?

Nishnha committed:
Add test for lockfile parsing to FileParser

Nishnha committed:
Pass filename to update_lockfile_declaration

baseballlover723 committed:
bump elixir version from 1.10.4 -> 1.11.4

cetinajero committed:
Bump to latest ruby versions

This includes:
  - 2.5.9
  - 2.6.7
  - 2.7.3
  - 3.0.1

Nishnha committed:
Modify FileParser#build_lockfile_dependency to update context in place

Nishnha committed:
Fix line endings for version.tf test

mctofu committed:
poetry: support pyproject.toml indentation

Nishnha committed:
Patch invalid registry source FileParser test

feelepxyz committed:
Merge pull request #3804 from baseballlover723/pr/elixir_1.11.4

bump elixir version from 1.10.4 -> 1.11.4

feelepxyz committed:
Merge pull request #3795 from dependabot/feelepxyz/composer-fix-git-clone-error

Composer: fix git clone error in lockfile updater

feelepxyz committed:
Merge pull request #3774 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/eslint-7.27.0

Bump eslint from 7.26.0 to 7.27.0 in /npm_and_yarn/helpers

jurre committed:
Tests: Rename non-existent go package name to domain we own

To prevent name squatting

feelepxyz committed:
Merge pull request #3805 from cetinajero/feature/bump-ruby-versions

Bump to latest ruby versions

feelepxyz committed:
Bundler: handle gemspec req ruby version ranges

Handle `required_ruby_version` ranges in gemspecs:
`required_ruby_version = ">= 2.2, < 4.0"` and
`required_ruby_version = Gem::Requirement.new(">= 2.1.8", "< 4.0.0")`

This would previously raise a `Gem::Requirement::BadRequirementError`.

Note: The `Bundler::Requirement` class splits requirements by `,`.

I've done an audit of other uses of `Gem::Requirement` but these seem ok
as we control the range we create so shouldn't run into similar bugs
there.

jurre committed:
Merge pull request #3808 from dependabot/jurre/clean-up-squattable-repos

Tests: Rename non-existent go package name to domain we own

feelepxyz committed:
Merge pull request #3809 from dependabot/feelepxyz/handle-ruby-version-ranges

Bundler: handle gemspec required ruby version ranges

feelepxyz committed:
v0.149.3

feelepxyz committed:
Merge pull request #3810 from dependabot/v0.149.3-release-notes

v0.149.3

Nishnha committed:
Update FileSelector#terragrunt_file? to use the lock_file? method

Co-authored-by: Jurre <jurre@github.com>

mctofu committed:
Merge pull request #3807 from dependabot/mctofu/poetry-indentation

poetry: support pyproject.toml indentation

xlgmokha committed:
test: ensure terraform HTTP API service discovery is performed

xlgmokha committed:
fix: perform service discovery

https://www.terraform.io/docs/internals/remote-service-discovery.html#discovery-process

xlgmokha committed:
chore: update vcr fixtures to include service discovery requests

xlgmokha committed:
fix: perform service discovery for provider API

xlgmokha committed:
fix: use service discovery to find metadata

xlgmokha committed:
style: fix linter errors

xlgmokha committed:
chore: remove unused code

xlgmokha committed:
refactor: delegate to instance method instead of variable

xlgmokha committed:
test: add service discovery to fixtures

xlgmokha committed:
test: stub service discovery lookup

dependabot[bot] committed:
build(deps): bump @npmcli/arborist in /npm_and_yarn/helpers

Bumps [@npmcli/arborist](https://github.com/npm/arborist) from 2.6.0 to 2.6.1.
- [Release notes](https://github.com/npm/arborist/releases)
- [Changelog](https://github.com/npm/arborist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/arborist/compare/v2.6.0...v2.6.1)

---
updated-dependencies:
- dependency-name: "@npmcli/arborist"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump jest in /npm_and_yarn/helpers

Bumps [jest](https://github.com/facebook/jest) from 27.0.1 to 27.0.3.
- [Release notes](https://github.com/facebook/jest/releases)
- [Changelog](https://github.com/facebook/jest/blob/master/CHANGELOG.md)
- [Commits](https://github.com/facebook/jest/compare/v27.0.1...v27.0.3)

---
updated-dependencies:
- dependency-name: jest
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump pipenv from 2018.11.26 to 2021.5.29 in /python/helpers

Bumps [pipenv](https://github.com/pypa/pipenv) from 2018.11.26 to 2021.5.29.
- [Release notes](https://github.com/pypa/pipenv/releases)
- [Changelog](https://github.com/pypa/pipenv/blob/master/CHANGELOG.rst)
- [Commits](https://github.com/pypa/pipenv/compare/v2018.11.26...v2021.5.29)

---
updated-dependencies:
- dependency-name: pipenv
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Bundler tests: Use git sources for organizations that we own

Bundler git dependencies have the possibility to execute code during the
installation phase, to ensure this does not happen in our test suite,
this updates any tests that pull in a git dependency to reference an
organization/account that we own.

This has the added benefit of protecting us against those repositories
being renamed, moved or deleted.

dependabot[bot] committed:
Bump composer/composer from 2.0.12 to 2.0.14 in /composer/helpers/v2

Bumps [composer/composer](https://github.com/composer/composer) from 2.0.12 to 2.0.14.
- [Release notes](https://github.com/composer/composer/releases)
- [Changelog](https://github.com/composer/composer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/composer/composer/compare/2.0.12...2.0.14)

Signed-off-by: dependabot[bot] <support@github.com>

xlgmokha committed:
refactor: extract method to build url

xlgmokha committed:
refactor: raise a DependabotError instead of RuntimeError

xlgmokha committed:
Update terraform/lib/dependabot/terraform/registry_client.rb

Co-authored-by: Pete Wagner <1559510+thepwagner@users.noreply.github.com>

xlgmokha committed:
fix: expand absolute urls in service discovery

xlgmokha committed:
test: relative and missing service urls

xlgmokha committed:
fix: raise error for unsupported schemes

xlgmokha committed:
refactor: replace usages of RuntimeError with DependabotError

xlgmokha committed:
test: collapse subject and before blocks

xlgmokha committed:
test: metadata endpoint is not found

xlgmokha committed:
test: follow redirect to metadata endpoint

xlgmokha committed:
Merge remote-tracking branch 'up/main' into updates-1480-service-discovery

xlgmokha committed:
test: reproduce bug that occurs when updating a module

xlgmokha committed:
fix: update regex to match on optional hostname declaration

xlgmokha committed:
style: fix linter errors

xlgmokha committed:
test: remove non-essential elements from fixture

xlgmokha committed:
fix: match provider source addresses with optional hostname

xlgmokha committed:
style: fix linter errors

jurre committed:
Merge pull request #3777 from dependabot/dependabot/composer/composer/helpers/v2/composer/composer-2.0.14

Bump composer/composer from 2.0.12 to 2.0.14 in /composer/helpers/v2

jurre committed:
Merge pull request #3819 from dependabot/jurre/bundler-git-fixtures

Bundler tests: Use git sources for organizations that we own

xlgmokha committed:
Update terraform/lib/dependabot/terraform/file_updater.rb

Co-authored-by: Jurre <jurre@github.com>

xlgmokha committed:
Merge pull request #3811 from xlgmokha/updates-1480-service-discovery

Perform Terraform API service discovery

xlgmokha committed:
Merge pull request #3821 from dependabot/updates-1480-private-modules

Detect optional hostname in module/provider source declaration

xlgmokha committed:
v0.149.4

xlgmokha committed:
Merge pull request #3825 from xlgmokha/v0.149.4-release-notes

v0.149.4

xlgmokha committed:
refactor: raise PrivateSourceAuthenticationFailure instead of DependabotError

xlgmokha committed:
fix: support github urls that are not pinned

e.g.

```json
{
  "name": "foo",
  "version": "0.0.0",
  "dependencies": {
    "express": "expressjs/express",
    "mocha": "mochajs/mocha#4727d357ea",
    "module": "user/repo#feature\/branch"
  }
}
```

* https://docs.npmjs.com/cli/v7/configuring-npm/package-json#github-urls

dependabot[bot] committed:
chore(deps-dev): update rubocop requirement from ~> 1.15.0 to ~> 1.16.0

Updates the requirements on [rubocop](https://github.com/rubocop/rubocop) to permit the latest version.
- [Release notes](https://github.com/rubocop/rubocop/releases)
- [Changelog](https://github.com/rubocop/rubocop/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rubocop/rubocop/compare/v1.15.0...v1.16.0)

---
updated-dependencies:
- dependency-name: rubocop
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump composer/composer in /composer/helpers/v2

Bumps [composer/composer](https://github.com/composer/composer) from 2.0.13 to 2.0.14.
- [Release notes](https://github.com/composer/composer/releases)
- [Changelog](https://github.com/composer/composer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/composer/composer/compare/2.0.13...2.0.14)

---
updated-dependencies:
- dependency-name: composer/composer
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

feelepxyz committed:
Merge pull request #3831 from dependabot/dependabot/composer/composer/helpers/v2/composer/composer-2.0.14

chore(deps): bump composer/composer from 2.0.13 to 2.0.14 in /composer/helpers/v2

feelepxyz committed:
Enable automerge on dependabot pull requests

Enable automerge on dependabot pull requests if the PR. This won't actually merge the PR, it will still require an approval from a team member.

feelepxyz committed:
Merge pull request #3830 from dependabot/dependabot/bundler/common/rubocop-tw-1.16.0

chore(deps-dev): update rubocop requirement from ~> 1.15.0 to ~> 1.16.0 in /common

feelepxyz committed:
Merge pull request #3816 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/jest-27.0.3

build(deps-dev): bump jest from 27.0.1 to 27.0.3 in /npm_and_yarn/helpers

feelepxyz committed:
Merge pull request #3815 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/npmcli/arborist-2.6.1

build(deps): bump @npmcli/arborist from 2.6.0 to 2.6.1 in /npm_and_yarn/helpers

feelepxyz committed:
Update .github/workflows/dependabot-automerge.yml

feelepxyz committed:
Update .github/workflows/dependabot-automerge.yml

feelepxyz committed:
Merge pull request #3832 from dependabot/feelepxyz/enable-automerge-on-dependabot-prs

Enable automerge on dependabot pull requests

feelepxyz committed:
Update dependabot-automerge.yml

feelepxyz committed:
Update dependabot-automerge.yml

feelepxyz committed:
Update dependabot-automerge.yml

feelepxyz committed:
Update dependabot-automerge.yml

feelepxyz committed:
Update dependabot-automerge.yml

feelepxyz committed:
Update dependabot-automerge.yml

feelepxyz committed:
Update dependabot-automerge.yml

feelepxyz committed:
Update dependabot-automerge.yml

dependabot[bot] committed:
build(deps): bump detect-indent in /npm_and_yarn/helpers

Bumps [detect-indent](https://github.com/sindresorhus/detect-indent) from 6.0.0 to 6.1.0.
- [Release notes](https://github.com/sindresorhus/detect-indent/releases)
- [Commits](https://github.com/sindresorhus/detect-indent/compare/v6.0.0...v6.1.0)

---
updated-dependencies:
- dependency-name: detect-indent
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #3814 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/detect-indent-6.1.0

build(deps): bump detect-indent from 6.0.0 to 6.1.0 in /npm_and_yarn/helpers

jurre committed:
Hex tests: Use git dependencies of repositories we own

To prevent these repositories from being squatted or from otherwise changing
under our feet in unexpected ways, prefer using repositories that we own
as git dependencies in our tests.

xlgmokha committed:
Merge pull request #3827 from xlgmokha/updates-1480-raise-authentication-error

Raise PrivateSourceAuthenticationFailure

xlgmokha committed:
v0.149.5

Nishnha committed:
Do not include lockfile in directory when updating a lockfile dependency

Nishnha committed:
Update tests to new provider version, reflect prefer_lockfile_source

Nishnha committed:
Install Terraform in docker shell

Co-authored-by: mo khan <xlgmokha@github.com>

Nishnha committed:
Ignore .terraform and .terraform.lock.hcl

Co-authored-by: mo khan <xlgmokha@github.com>

Nishnha committed:
Add file selectors for lockfiles

Co-authored-by: mo khan <xlgmokha@github.com>

Nishnha committed:
Create lockfile fixture

Co-authored-by: mo khan <xlgmokha@github.com>

xlgmokha committed:
Merge pull request #3836 from dependabot/v0.149.5-release-notes

v0.149.5

Nishnha committed:
Fix FileUpdater test

Nishnha committed:
Create versions.tf, remove versioning from main.tf

Nishnha committed:
Pin lockfile version and constraint, add provider dependency

Nishnha committed:
Update all lockfile dependencies with `terraform init -upgrade`

Co-authored-by: mo khan <xlgmokha@github.com>

Nishnha committed:
Create test for lockfile update

dependabot[bot] committed:
build(deps): bump @npmcli/arborist in /npm_and_yarn/helpers

Bumps [@npmcli/arborist](https://github.com/npm/arborist) from 2.6.1 to 2.6.2.
- [Release notes](https://github.com/npm/arborist/releases)
- [Changelog](https://github.com/npm/arborist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/arborist/compare/v2.6.1...v2.6.2)

---
updated-dependencies:
- dependency-name: "@npmcli/arborist"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Add retry for Azure client POST

dependabot[bot] committed:
build(deps-dev): bump friendsofphp/php-cs-fixer in /composer/helpers/v2

Bumps [friendsofphp/php-cs-fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) from 2.19.0 to 3.0.0.
- [Release notes](https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases)
- [Changelog](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3.0/CHANGELOG.md)
- [Commits](https://github.com/FriendsOfPHP/PHP-CS-Fixer/compare/v2.19.0...v3.0.0)

---
updated-dependencies:
- dependency-name: friendsofphp/php-cs-fixer
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #3839 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/npmcli/arborist-2.6.2

build(deps): bump @npmcli/arborist from 2.6.1 to 2.6.2 in /npm_and_yarn/helpers

jurre committed:
Merge pull request #3835 from dependabot/jurre/hex-git-deps

Hex tests: Use git dependencies of repositories we own

Nishnha committed:
Add expected lockfile for hashicorp/random

Nishnha committed:
Parse provider source strings

Nishnha committed:
Update lockfiles when provider dependencies are updated

Commit:
Use double quotes to avoid extra backslashes for escaping.

Nishnha committed:
Check lockfile exists before adding to updated_files, fix short-circuit bug

Co-authored-by: Jurre Stender <jurre@github.com>

dependabot[bot] committed:
build(deps): bump dependabot/fetch-metadata from 1.0.2 to 1.0.3

Bumps [dependabot/fetch-metadata](https://github.com/dependabot/fetch-metadata) from 1.0.2 to 1.0.3.
- [Release notes](https://github.com/dependabot/fetch-metadata/releases)
- [Commits](https://github.com/dependabot/fetch-metadata/compare/v1.0.2...v1.0.3)

---
updated-dependencies:
- dependency-name: dependabot/fetch-metadata
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #3848 from dependabot/dependabot/github_actions/dependabot/fetch-metadata-1.0.3

build(deps): bump dependabot/fetch-metadata from 1.0.2 to 1.0.3

Nishnha committed:
Update lockfile dependencies one at a time using `terraform providers lock`

Nishnha committed:
Add codepath for source type 'lockfile' in FileUpdater

Nishnha committed:
Merge get(endpoint:) function from main

Nishnha committed:
Update tests to reflect latest version

Nishnha committed:
Add UpdateChecker#lockfile_dependency?

Nishnha committed:
Modify registry_source_details_from

Nishnha committed:
Ensure the lockfile version changes

Co-authored-by: Jurre Stender <jurre@github.com>

Nishnha committed:
Rubocop linting

dependabot[bot] committed:
build(deps-dev): bump prettier in /npm_and_yarn/helpers

Bumps [prettier](https://github.com/prettier/prettier) from 2.3.0 to 2.3.1.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.3.0...2.3.1)

---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint in /npm_and_yarn/helpers

Bumps [eslint](https://github.com/eslint/eslint) from 7.27.0 to 7.28.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.27.0...v7.28.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump composer/composer in /composer/helpers/v2

Bumps [composer/composer](https://github.com/composer/composer) from 2.0.14 to 2.1.1.
- [Release notes](https://github.com/composer/composer/releases)
- [Changelog](https://github.com/composer/composer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/composer/composer/compare/2.0.14...2.1.1)

---
updated-dependencies:
- dependency-name: composer/composer
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

feelepxyz committed:
Delete automerge until PR state issue is fixed

feelepxyz committed:
Merge pull request #3853 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/eslint-7.28.0

build(deps-dev): bump eslint from 7.27.0 to 7.28.0 in /npm_and_yarn/helpers

feelepxyz committed:
Merge pull request #3852 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/prettier-2.3.1

build(deps-dev): bump prettier from 2.3.0 to 2.3.1 in /npm_and_yarn/helpers

dependabot[bot] committed:
build(deps-dev): bump jest in /npm_and_yarn/helpers

Bumps [jest](https://github.com/facebook/jest) from 27.0.3 to 27.0.4.
- [Release notes](https://github.com/facebook/jest/releases)
- [Changelog](https://github.com/facebook/jest/blob/master/CHANGELOG.md)
- [Commits](https://github.com/facebook/jest/compare/v27.0.3...v27.0.4)

---
updated-dependencies:
- dependency-name: jest
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

feelepxyz committed:
Rename config file

feelepxyz committed:
Fix error message matches

feelepxyz committed:
Merge pull request #3846 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/jest-27.0.4

build(deps-dev): bump jest from 27.0.3 to 27.0.4 in /npm_and_yarn/helpers

github-actions[bot] committed:
Merge pull request #3854 from dependabot/dependabot/composer/composer/helpers/v2/composer/composer-2.1.1

build(deps): bump composer/composer from 2.0.14 to 2.1.1 in /composer/helpers/v2

feelepxyz committed:
v0.150.0

feelepxyz committed:
Fix nuget spec

feelepxyz committed:
Merge pull request #3858 from dependabot/v0.150.0-release-notes

v0.150.0

github-actions[bot] committed:
Merge pull request #3840 from dependabot/dependabot/composer/composer/helpers/v2/friendsofphp/php-cs-fixer-3.0.0

build(deps-dev): bump friendsofphp/php-cs-fixer from 2.19.0 to 3.0.0 in /composer/helpers/v2

mctofu committed:
Pin erlang to OTP 23 until we can resolve OTP 24 warning issues

xlgmokha committed:
test: pin assertions to the latest HEAD commit

Nishnha committed:
Remove copy_dir_to_temp_directory, use SharedHelpers inline

Nishnha committed:
Revert dependabot/FileUpdaters/base changes, reimplement in terraform/FileUpdater

Nishnha committed:
Revert changes to registry_source_details_from

There is a separate function to get provider source details. This commit removes checks for provider sources from the registry source details function

mctofu committed:
Merge pull request #3862 from dependabot/mctofu/pin-erlang

Pin erlang to OTP 23 until we can resolve OTP 24 warning issues

mctofu committed:
v0.151.0

mctofu committed:
Merge pull request #3863 from dependabot/v0.151.0-release-notes

v0.151.0

Nishnha committed:
Memoize update_lockfile_declaration method by dependency

xlgmokha committed:
Merge pull request #3837 from xlgmokha/core-3813

fix(npm): Prevent unnecessary hash pinning in lock file constraint

Nishnha committed:
Rubocop

xlgmokha committed:
v0.151.1

xlgmokha committed:
Merge pull request #3864 from xlgmokha/v0.151.1-release-notes

v0.151.1

dependabot[bot] committed:
build(deps): bump composer/composer in /composer/helpers/v2

Bumps [composer/composer](https://github.com/composer/composer) from 2.1.1 to 2.1.2.
- [Release notes](https://github.com/composer/composer/releases)
- [Changelog](https://github.com/composer/composer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/composer/composer/compare/2.1.1...2.1.2)

---
updated-dependencies:
- dependency-name: composer/composer
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #3866 from dependabot/dependabot/composer/composer/helpers/v2/composer/composer-2.1.2

jurre committed:
Merge pull request #3843 from AlekhyaYalla/alekhyayalla/retry_azure_post_request

[Azure] Add retries for Azure client POST

Nishnha committed:
Remove lockfile_dependency? from UpdateChecker

Nishnha committed:
Revert consul/aws invalid module test

Nishnha committed:
Revert splitting case statement

Nishnha committed:
Remove dependency argument from update_lockfile_declaration

Nishnha committed:
Remove checks for lockfile in updated_dependency_files

Nishnha committed:
Remove lockfile_changed? method

Nishnha committed:
Remove memoization for update_lockfile_declaration

Nishnha committed:
Move lock_file.empty? check to update_lockfile_declaration method

Nishnha committed:
Replace FileSelector#lock_file select with find

Co-authored-by: Philip Harrison <feelepxyz@github.com>

Nishnha committed:
Refactor FileFetcher#lock_file

Co-authored-by: Philip Harrison <feelepxyz@github.com>

Nishnha committed:
Remove RegistryClient#get

Co-authored-by: Philip Harrison <feelepxyz@github.com>

Nishnha committed:
Invert updated_content guard

Co-authored-by: Philip Harrison <philip@mailharrison.com>

jurre committed:
Drop python 2.x support and fix pipenv 2021.5.29 tests and

Python 2 has been unsupported for a year and a half now, pip 21 no
longer supports it and we see pretty small usage numbers at GitHub.
Maintaining support for it has become increasingly harder, and this
pipenv upgrade made it even more cumbersome.

It's been a good ride, python 2. See ya!

Nishnha committed:
Add a test to check updated_dependency_files doesn't return a lockfile if it doesn't change

Co-authored-by: Philip Harrison <feelepxyz@github.com>

Nishnha committed:
Rubocop, remove commented out code

jurre committed:
Merge pull request #3817 from dependabot/dependabot/pip/python/helpers/pipenv-2021.5.29

build(deps): bump pipenv from 2018.11.26 to 2021.5.29 in /python/helpers

jurre committed:
Upgrade pip to 21.1.2 and pip-tools to 6.1.0

https://github.com/jazzband/pip-tools/blob/master/CHANGELOG.md#610-2021-04-14

https://pip.pypa.io/en/stable/news/#v21-1-2

Nishnha committed:
Pin Terraform version

Nishnha committed:
Merge pull request #3766 from dependabot/nishnha/terraform-lockfile-support

Terraform: Add lockfile update support

Nishnha committed:
Add lockfile as an array to file_fetcher

Nishnha committed:
Add lockfile test to FileFetcher

jurre committed:
Merge pull request #3871 from dependabot/jurre/pip21-pip-tools-6.1

Nishnha committed:
Merge pull request #3872 from dependabot/nishnha/terraform-lockfile-fix

Fix Terraform lockfile bug introduced in #3766

jurre committed:
v0.152.0

jurre committed:
Merge pull request #3879 from dependabot/v0.152.0-release-notes

v0.152.0

GiriB commented on npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker.rb:
```diff
@@ -297,9 +297,8 @@ def library?
-
-        raise "Multiple sources! #{sources.join(', ')}" if sources.count > 1
+          dependency.requirements.map { |r| r.fetch(:source) }.uniq.compact.
+          sort_by { |source| RegistryFinder.central_registry?(source[:url]) ? 1 : 0 }
```
Should not effect us right?

Nishnha committed:
Add instruction to checkout new branch

Nishnha committed:
DependencyFileNotParseable error with useful message for old terraform  provider syntax

Nishnha committed:
Check if old provider syntax using is_a? instead of respond_to?

Nishnha committed:
Add test case for hcl1 provider syntax

milind009 commented on Dockerfile:
```diff
@@ -70,7 +70,7 @@ RUN apt-add-repository ppa:brightbox/ruby-ng \
   && gem update --system 3.2.14 \
   && gem install bundler -v 1.17.3 --no-document \
-  && gem install bundler -v 2.2.15 --no-document \
+  && gem install bundler -v 2.2.18 --no-document \
```
@AlekhyaYalla we will have to make a similar change in our Dockerfile.lite

milind009 commented on common/lib/dependabot/config.rb:
```diff
@@ -0,0 +1,7 @@
+
+module Dependabot
+  module Config
+    class InvalidConfigError < StandardError; end
```
nit: we could use something like this for tracking errors due to incorrect checkin in dmaas.yml

milind009 replied:
Not essentially in this PR but something to keep in mind

jurre committed:
Tests: Allow profiling tests with stackprof when tagged

This adds the ability to tag a spec with `:profile`:

```ruby
it "profiles with rspec", :profile do
  # ..
end
```

And will then generate a stackprof dump in the `tmp` directory.

This is useful when debugging performance issues.

jurre committed:
Terraform: patch up tests

These tests pull in the latest version of `hashicorp/aws`, which just
updated.

We should look into replacing this to use a provider that we own, so
that we don't end up having to keep up with whatever aws is releasing.

jurre committed:
Merge pull request #3885 from dependabot/jurre/patch-tf-tests

Terraform: patch up tests

jurre committed:
Merge pull request #3884 from dependabot/jurre/rspec-profile

Tests: Allow profiling tests with stackprof when tagged

Nishnha committed:
Modify fixture name

Nishnha committed:
Merge pull request #3882 from dependabot/update-bump-version

Add instruction to checkout new branch

Nishnha committed:
Merge pull request #3883 from dependabot/nishnha/terraform-depricated-warning

Terraform: Helpful error message for old provider syntax

Nishnha committed:
v0.152.1

Nishnha committed:
Merge pull request #3886 from dependabot/v0.152.1-release-notes

v0.152.1
